core/trino-main/src/test/java/io/trino/server/security/TestResourceSecurity.java

```diff
@@ -538,6 +552,34 @@ public void testJwtWithJwkAuthenticator()
         }
     }
 
+    @Test
+    public void testJwtAuthenticatorWithInvalidAudience()
```

Comment:
I think we're missing a test with empty audience and without `http-server.authentication.jwt.required-audience` specified.